src/lib/reporting.ml

```diff
@@ -168,6 +168,7 @@ type error =
   | Err_syntax_loc of Parse_ast.l * string
   | Err_lex of Lexing.position * string
   | Err_type of Parse_ast.l * string option * string
+  | Err_smt_failure of Parse_ast.l * string
```

Comment:
I think we can probably just use `Err_general` for this?

Reply:
Good idea.